cmd/completion.go

```diff
@@ -48,7 +51,13 @@ MacOS:
   $ weep completion bash > /usr/local/etc/bash_completion.d/weep
 `,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return cmd.Root().GenBashCompletion(os.Stdout)
+		logging.Log.WithFields(logrus.Fields{"shell": "bash"}).Infoln("Generate completion")
```

Comment:
You can dedupe these by moving the log call up to the `completionCommand` pre-run:

```go
PersistentPreRun: func(cmd *cobra.Command, args []string) {
	logging.Log.WithFields(logrus.Fields{"shell": cmd.CalledAs()}).Infoln("Generate completion")
},
```

Reply:
Ooh interesting, I didn't realize the `PersistentPreRun` would stomp on the root one. Looks like I've got some reading to do. :eyes: